pypeit/reduce.py

```diff
@@ -314,105 +409,104 @@ def global_skysub(self, std=False, skymask=None, update_crmask=True, maskslits=N
-                                                             no_poly=self.redux_par['no_poly'],
-                                                             pos_mask = (not self.ir_redux),
+                                                             bsp=self.par['scienceimage']['skysub']['bspline_spacing'],
+                                                             no_poly=self.par['scienceimage']['skysub']['no_poly'],
+                                                             pos_mask=(not self.ir_redux),
                                                              show_fit=show_fit)
             # Mask if something went wrong
             if np.sum(self.global_sky[thismask]) == 0.:
                 self.maskslits[slit] = True
 
         if update_crmask:
             self.sciImg.update_mask_cr(subtract_img=self.global_sky)
-            #self.crmask = procimg.lacosmic(self.det, self.sciimg-self.global_sky,
-            #                               self.spectrograph.detector[self.det-1]['saturation'],
-            #                               self.spectrograph.detector[self.det-1]['nonlinear'],
-            #                               varframe=utils.calc_ivar(self.sciivar),
-            #                               maxiter=self.proc_par['lamaxiter'],
-            #                               grow=self.proc_par['grow'],
-            #                               remove_compact_obj=self.proc_par['rmcompact'],
-            #                               sigclip=self.proc_par['sigclip'],
-            #                               sigfrac=self.proc_par['sigfrac'],
-            #                               objlim=self.proc_par['objlim'])
-            # Rebuild the mask with this new crmask
-            #self.mask = procimg.update_mask_cr(self.sciImg.bitmask, self.mask, self.crmask)
 
         # Step
         self.steps.append(inspect.stack()[0][3])
 
         if show:
             sobjs_show = None if show_objs else self.sobjs_obj
             # Global skysub is the first step in a new extraction so clear the channels here
-            self.show('global', slits=True, sobjs =sobjs_show, clear=False)
+            self.show('global', slits=True, sobjs=sobjs_show, clear=False)
 
         # Return
         return self.global_sky
 
     def local_skysub_extract(self, waveimg, global_sky, sobjs,
-                             maskslits=None, model_noise=True, std=False,
+                             model_noise=True, spat_pix=None,
                              show_profile=False, show_resids=False, show=False):
-
         """
-         Dummy method for locak skysubtraction and extraction. Overloaded by class specific skysub and extraction.
+        Dummy method for locak skysubtraction and extraction.
 
-         Returns:
+        Overloaded by class specific skysub and extraction.
 
-         """
+        Returns:
 
-        return None, None, None, None, None
+        """
 
+        return None, None, None, None, None
 
```

Comment:
Should these methods flexure and helio be in reduce or somewhere else?

Reply:
Also should these be class methods?

Reply:
I think they should be in `Reduce` as class methods.

Reply:
done.